### ISSUE TITLE
Negative adjacency matrix

### ISSUE THREAD

**Issue by MaKaili** [chaitjo/learning-tsp#2]
Hi @chaitjo, why do you use a negative adjacency matrix? It seems that your implementation of the encoder is agnostic to positive or negative adjacent matrix since you use `nn.Embedding(2, hidden_dim)` to represent edge existing or not. Could you share your motivation on negative adjacent matrix? Thanks in advance.

**Comment by chaitjo** [comment]
Hi @MaKaili, thank you for your interest! It's been a while since I ran this code, so can you point me to some specific lines/functions? Happy to help but may need some more context.

**Comment by MaKaili** [comment]
Thanks for your kindly reply.
It is the function `nearest_neighbor_graph` in
https://github.com/chaitjo/learning-tsp/blob/a7b132aa4e532b3f23e74c240c7bfbb794ebf764/problems/tsp/problem_tsp.py#L13-L35

**Comment by chaitjo** [comment]
https://github.com/chaitjo/learning-tsp/blob/a7b132aa4e532b3f23e74c240c7bfbb794ebf764/nets/encoders/gnn_encoder.py#L117

You can see that the graph structure is enforced through masking in this particular implementation. Thus, in the mask, 1 --> pairs of nodes for which we mask out the messages; 0 --> do not mask. 

Does that help?

**Comment by MaKaili** [comment]
Yes, thanks for the reply.

**Comment by chaitjo** [comment]
Great, happy to help further.
